mitiq/zne/inference.py

```diff
@@ -157,6 +158,57 @@ def mitiq_polyfit(
     return list(opt_params)
 
 
+def _make_batched_executor(
+        executor: Union[Callable[..., float], Callable[..., List[float]]],
+) -> Callable[..., List[float]]:
+    """Function transforming a single-circuit executor into a batched executor.
+    If the input is annotated with a return type that is one of the following:
+        * Iterable[float]
+        * List[float]
+        * Sequence[float]
+        * Tuple[float]
+        * numpy.ndarray
+    it is detected as "batched" and is returned without any change.
+
+    Note: Once a single-circuit executor is transformed into a batched one,
+    keyword arguments can be passed using the optional argument
+    (batched_kwargs) in the form of a list of ditionaries.
+
+    Args:
+        executor: A "single executor" (1) or a "batched executor" (2).
+            (1) A function which inputs a single circuit and outputs a
+            single expectation value of interest.
+            (2) A function which inputs a list of circuits and outputs a
+            list of expectation values (one for each circuit).
+
+    Returns:
+        batched_executor: The batched executor.
+    """
+    executor_annotation = inspect.getfullargspec(executor).annotations
+    if executor_annotation.get("return") in (
+        List[float],
+        Sequence[float],
+        Tuple[float],
+        Iterable[float],
+        np.ndarray,
+    ):
+        return executor  # type: ignore
+
+    @wraps(executor)
+    def batched_executor(
+        batch: List[QPROGRAM], batched_kwargs: Optional[List[Dict]] = None,
+    ) -> List[float]:
+
+        if batched_kwargs is None:
+            return [executor(circ) for circ in batch]
+        results = []
+        for circ, kwargs in zip(batch, batched_kwargs):
+            results.append(executor(circ, **kwargs))
+        return results  # type: ignore
```

Comment:
Also can be simplified by allowing `batched_kwargs` to be a list of empty dictionaries. See comment in `_get_keyword_args`.